src/dirs.rs

```diff
@@ -1,37 +1,38 @@
-// We define some unused constants here, since we don't have compile time string concatination.
-#![allow(dead_code)]
+use std::env;
+use std::ffi::OsStr;
+use std::path::PathBuf;
 
-pub const WORK_DIR: &'static str = "./work";
-pub const LOCAL_DIR: &'static str = "./work/local";
+lazy_static! {
+    pub static ref WORK_DIR: PathBuf = {
+        env::var_os("CARGOBOMB_WORK")
+            .unwrap_or_else(|| OsStr::new("./work").to_os_string())
+            .into()
+    };
+    pub static ref LOCAL_DIR: PathBuf = WORK_DIR.join("local");
 
-pub const CARGO_HOME: &'static str = "./work/local/cargo-home";
-pub const RUSTUP_HOME: &'static str = "./work/local/rustup-home";
+    pub static ref CARGO_HOME: String = LOCAL_DIR.join("cargo-home").to_string_lossy().into();
+    pub static ref RUSTUP_HOME: String = LOCAL_DIR.join("rustup-home").to_string_lossy().into();
```

Comment:
These should be `PathBuf` too, but the logic in `toolchain.rs` is annoying to work with, since it expects strings in a bunch of places.